File: service/src/test/kotlin/org/eclipse/keti/acs/config/PolicyConditionCacheConfigTest.java
package org.eclipse.keti.acs.config;

import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.core.IsInstanceOf.instanceOf;

import org.eclipse.keti.acs.commons.policy.condition.groovy.InMemoryGroovyConditionCache;
import org.eclipse.keti.acs.commons.policy.condition.groovy.NonCachingGroovyConditionCache;
import org.testng.annotations.Test;

public class PolicyConditionCacheConfigTest {

    private PolicyConditionCacheConfig policyConditionCacheConfig = new PolicyConditionCacheConfig();

    @Test
    public void testPolicyConditionCacheConfigDisabled() {
        assertThat(this.policyConditionCacheConfig.conditionCache(true),
                instanceOf(NonCachingGroovyConditionCache.class));
    }

    @Test
    public void testPolicyConditionCacheConfigEnabled() {
        assertThat(this.policyConditionCacheConfig.conditionCache(false),
                instanceOf(InMemoryGroovyConditionCache.class));
    }
}

File: service/src/test/kotlin/org/eclipse/keti/acs/config/PolicyConditionCacheConfigTest.kt
package org.eclipse.keti.acs.config

import org.eclipse.keti.acs.commons.policy.condition.groovy.InMemoryGroovyConditionCache
import org.eclipse.keti.acs.commons.policy.condition.groovy.NonCachingGroovyConditionCache
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.core.IsInstanceOf.instanceOf
import org.testng.annotations.Test

class PolicyConditionCacheConfigTest {

    private val policyConditionCacheConfig = PolicyConditionCacheConfig()

    @Test
    fun testPolicyConditionCacheConfigDisabled() {
        assertThat(
            this.policyConditionCacheConfig.conditionCache(true),
            instanceOf(NonCachingGroovyConditionCache::class.java)
        )
    }

    @Test
    fun testPolicyConditionCacheConfigEnabled() {
        assertThat(
            this.policyConditionCacheConfig.conditionCache(false),
            instanceOf(InMemoryGroovyConditionCache::class.java)
        )
    }
}
